src/data-adapter.js

```diff
@@ -14,9 +14,20 @@ const parseDirPath = (dirPath) => {
 const removeDuplicates = arr => Array.from(new Set(arr));
 
 export default function getDataAdapterOptions(data, savePath = process.cwd()) {
-  return removeDuplicates(data).map((urlString) => ({
-    host: url.parse(urlString).host,
-    path: url.parse(urlString).path,
-    fileName: `${parseDirPath(savePath)}${path.basename(urlString)}`
-  }));
+  return removeDuplicates(data).map((item) => {
+    const itemurl = (typeof item === 'string') ? item : item.url;
```

Comment:
Are we 100% sure that `item` would be either a string or an object with an url property?